packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,408 @@
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
+        txConfirmations: pendingDeposit?.transactionConfirmations,
+        receivedAt: swapRequest?.depositReceivedAt?.valueOf(),
+        receivedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+      },
+      swap: {
+        ...sortedSwaps?.reduce(
+          (acc, curr) => {
+            if (isEgressableSwap(curr)) {
+              acc.totalAmountSwapped = acc.totalAmountSwapped.plus(curr.swapOutputAmount ?? 0);
+              if (curr.swapScheduledAt && curr.swapExecutedAt) {
+                acc.lastExecutedChunk = getSwapFields(curr);
+                acc.totalChunksExecuted += 1;
+              }
+            }
+            return acc;
+          },
+          {
+            totalAmountSwapped: new Prisma.Decimal(0),
+            totalChunksExecuted: 0,
+            currentChunk: sortedSwaps[0],
+            lastExecutedChunk: null as null | ReturnType<typeof getSwapFields>,
+            isDcaSwap:
+              swapDepositChannel?.chunkIntervalBlocks && swapDepositChannel.chunkIntervalBlocks > 1,
+          },
+        ),
+        egress: {
+          amount: swapEgress?.amount?.toFixed(),
+          scheduledAt: swapEgress?.scheduledAt?.valueOf(),
+          scheduledBlockIndex: swapEgress?.scheduledBlockIndex ?? undefined,
+          ignoredAt: swapRequest?.ignoredEgresses
+            .find((e) => e.type === 'SWAP')
+            ?.ignoredAt?.valueOf(),
+          ignoredBlockIndex: swapRequest?.ignoredEgresses.find((e) => e.type === 'SWAP')
+            ?.ignoredBlockIndex,
+          ignoredAmount: swapRequest?.ignoredEgresses
+            .find((e) => e.type === 'SWAP')
+            ?.amount?.toFixed(),
+        },
+        broadcast: {
+          requestedAt: swapEgress?.broadcast?.requestedAt?.valueOf(),
+          requestedBlockIndex: swapEgress?.broadcast?.requestedBlockIndex ?? undefined,
+          abortedAt: swapEgress?.broadcast?.abortedAt?.valueOf(),
+          abortedBlockIndex: swapEgress?.broadcast?.abortedBlockIndex ?? undefined,
+          succeededAt: swapEgress?.broadcast?.succeededAt?.valueOf(),
+          succeededBlockIndex: swapEgress?.broadcast?.succeededBlockIndex ?? undefined,
+          txRef: egress?.broadcast?.transactionRef ?? egressTrackerTxRef,
+        },
+        srcChainRequiredBlockConfirmations:
+          (internalSrcAsset && (await getRequiredBlockConfirmations(internalSrcAsset))) ??
+          undefined,
+        estimatedDefaultDurationSeconds:
+          srcAsset && destAsset && (await estimateSwapDuration({ srcAsset, destAsset })),
+      },
+      refund: {
+        egress: {
```

Comment:
would probably flatten this a bit and remove `egress` and `broadcast`. I think we exposed too many internal implementation details in the original one. Instead we can do something more like:
```ts
refund?: {
  scheduledAt: Date;
  scheduledAtBlockIndex: string;
  failedAt?: Date;
  failedAtBlockIndex?: string;
  failureReason?: string;
  completedAt?: Date;
  completedAtBlockIndex?: string;
}
```

if the broadcast is aborted or the egress is ignored, we can just use the `failedAt` field and included failure info to describe the specific mode of failure.